blueocean-dashboard/src/main/js/components/PageLoading.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { Progress } from '@jenkins-cd/design-language';
+
+/**
+ * Simple component to render a loading animation
+ */
+export default class PageLoading extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            percentage: undefined,
+            running: false,
+        };
+    }
+    
+    componentWillMount() {
+        if (this.props.duration) {
```

Comment:
What exactly is `duration`? The name suggests it's the total length of time for which the progress bar remains visible (the 100%), but I'm not sure from the code.